manager/integration/tests/test_backing_image.py

```diff
@@ -694,3 +704,199 @@ def test_volume_wait_for_backing_image_condition(client): # NOQA
     volume_endpoint = get_volume_endpoint(volume2)
     vol2_cksum = get_device_checksum(volume_endpoint)
     assert vol1_cksum == vol2_cksum
+
+
+@pytest.mark.backing_image  # NOQA
+def test_backing_image_min_number_of_replicas(client): # NOQA
+    """
+    Test the backing image minNumberOfCopies
+
+    Given
+    - Create a BackingImage
+    - Update the minNumberOfCopies to 2
+
+    When
+    - The BackingImage is prepared and transferred to manager
+
+    Then
+    - There will be two ready file of the BackingIamge
+
+    Given
+    - Set the setting backing-image-cleanup-wait-interval to 1 minute
+
+    When
+    - Update the minNumberOfCopies to 1
+    - Wait for 1 minute
+
+    Then
+    - There will be one ready BackingImage file left.
+    """
+    bi_url = BACKING_IMAGE_RAW_URL
+    create_backing_image_with_matching_url(
+        client, BACKING_IMAGE_NAME, bi_url)
+
+    backing_image = client.by_id_backing_image(BACKING_IMAGE_NAME)
+    backing_image = backing_image.updateMinNumberOfCopies(minNumberOfCopies=2)
+
+    check_backing_image_disk_map_status(client, BACKING_IMAGE_NAME, 2, "ready")
+
+    update_setting(client, "backing-image-cleanup-wait-interval", "1")
+
+    backing_image = client.by_id_backing_image(BACKING_IMAGE_NAME)
+    backing_image = backing_image.updateMinNumberOfCopies(minNumberOfCopies=1)
+
+    time.sleep(60)
+
+    check_backing_image_disk_map_status(client, BACKING_IMAGE_NAME, 1, "ready")
```

Comment:
Can it be replaced a polling for loop such as
```
max_retries = 60
for (i = 0; i < max_retries; i++)
  check_backing_image_disk_map_status(client, BACKING_IMAGE_NAME, 1, "ready")
  if i == max_retries - 1
    raise error
```

Reply:
`check_backing_image_disk_map_status()` itself is a for loop polling already
```
def check_backing_image_disk_map_status(client, bi_name, expect_cnt, expect_disk_state): # NOQA
    # Number of expect_cnt should equal to number of disk map
    # that have expect_disk_state

    for i in range(RETRY_COUNTS):
        backing_image = client.by_id_backing_image(bi_name)

        count = 0
        for disk_id, status in iter(backing_image.diskFileStatusMap.items()):
            if status.state == expect_disk_state:
                count = count + 1

        if expect_cnt == count:
            break
        time.sleep(RETRY_INTERVAL)

    assert expect_cnt == count
```